almkanal/almkanal.py

```diff
@@ -0,0 +1,629 @@
+        ica_lp_freq : float | None, optional
+            Low-pass filter frequency for ICA preprocessing. Defaults to None.
+        resample_freq : int, optional
+            Downsampling frequency before ICA. Defaults to 200 Hz.
+        eog : bool, optional
+            Whether to detect and remove EOG artifacts. Defaults to True.
+        eog_corr_thresh : float, optional
+            Correlation threshold for EOG artifact detection. Defaults to 0.5.
+        ecg : bool, optional
+            Whether to detect and remove ECG artifacts. Defaults to True.
+        ecg_corr_thresh : float, optional
+            Correlation threshold for ECG artifact detection. Defaults to 0.5.
+        emg : bool,
+            Whether to detect and remove EMG artifacts. Defaults to False.
+        emg_thresh:
+            Value above which a component should be marked as muscle-related, relative to a typical muscle component.
+        train : bool, optional
+            Whether to detect and remove train-related artifacts. Defaults to True.
+        train_freq : int, optional
+            Frequency for train artifact detection. Defaults to 16 Hz.
+        img_path : str | None, optional
+            Path to save ICA plots. Defaults to None.
+        fname : str | None, optional
+            Filename for ICA plots. Defaults to None.
+
+        Returns
+        -------
+        None
+        """
+
+        # this should do an ica
+        ica_info = ICAInfoDict(
+            n_components=n_components,
+            method=method,
+            random_state=random_state,
+            fit_params=fit_params,
+            ica_hp_freq=ica_hp_freq,
+            ica_lp_freq=ica_lp_freq,
+            resample_freq=resample_freq,
+            eog=eog,
+            eog_corr_thresh=eog_corr_thresh,
+            surrogate_eog_chs=surrogate_eog_chs,
+            ecg=ecg,
+            ecg_corr_thresh=ecg_corr_thresh,
+            emg=emg,
+            emg_thresh=emg_thresh,
+            train=train,
+            train_freq=train_freq,
+            train_thresh=train_thresh,
+            img_path=img_path,
+            fname=fname,
+        )
+
+        if self.info.ica is None:
+            self.info.ica = [ica_info]
+
+            self.raw, ica, ica_ids = run_ica(self.raw, **ica_info)
+            self.ica = [ica]
+            self.ica_ids = [ica_ids]
+        elif self.info.ica is not None:
+            assert isinstance(self.info.ica, list)
+            # Take care of case where you ran multiple icas.
+            # TODO: When we end up applying them to the noise cov dont forget
+            # to also do it successively.
+            self.info.ica.append(ica_info)
+
+            self.raw, ica, ica_ids = run_ica(self.raw, **ica_info)
+            assert isinstance(self.ica, list)
+            self.ica.append(ica)
+            assert isinstance(self.ica_ids, list)
+            self.ica_ids.append(ica_ids)
+
+    def do_bio_process(
```

Comment:
I really like those signals to be extra channels of the raw object.

And I think, that `do` is not a good verb in this API. I would suggest something like `compute`, `apply`.....